docs: Add missing authentication and advanced configs for Druid connector (#27611)
prestodb/presto: presto-docs/src/main/sphinx/connector/druid.rst

``case-sensitive-name-matching``         Enable case-sensitive identifier support for schema,
                                         table, and column names for the connector. When disabled,
                                         names are matched case-insensitively using lowercase
                                         normalization. Default is ``false``.
``druid.tls.enabled``                    Enable TLS when connecting to Druid.
``druid.tls.truststore-path``            Path to the trust certificate file.
``druid.tls.truststore-password``        Password for the trust certificate file.
``druid.authentication.type``            Authentication type for Druid.
``druid.basic.authentication.username``  Username for basic authentication.
``druid.basic.authentication.password``  Password for basic authentication.
``druid.hadoop.config.resources``        Hadoop configuration resources.
``druid.ingestion.storage.path``         Local storage path for ingestion.
======================================== =========================================================

``druid.coordinator-url``
^^^^^^^^^^^^^^^^^^^^^^^^^

> Druid coordinator URL, e.g. localhost:8081.

Reviewer: ```suggestion
Druid coordinator URL. For example, ``localhost:8081``.
```
Avoid Latin abbreviations. See the [GitLab documentation style guide recommended word list entry for e.g.](https://docs.gitlab.com/development/documentation/styleguide/word_list/#eg) for discussion and alternatives.

> Druid coordinator URL, For example, ``localhost:8081``.

Reviewer: ```suggestion
Druid coordinator URL. For example, ``localhost:8081``.
```

``druid.broker-url``
^^^^^^^^^^^^^^^^^^^^

> Druid broker URL, e.g. localhost:8082.

Reviewer: ```suggestion
Druid broker URL. For example, ``localhost:8082``.
```

> Druid broker URL, For example, ``localhost:8082``.

Reviewer: ```suggestion
Druid broker URL. For example, ``localhost:8082``.
```

Password for the trust certificate file.

``druid.authentication.type``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Authentication type for Druid.

Supported values are: ``NONE`` (default), ``BASIC`` and ``KERBEROS``.

``druid.basic.authentication.username``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Username for basic authentication. Required when ``druid.authentication.type`` is set to ``BASIC``.

``druid.basic.authentication.password``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Password for basic authentication. Required when ``druid.authentication.type`` is set to ``BASIC``.

``druid.hadoop.config.resources``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Comma-separated list of Hadoop configuration resource paths (e.g., ``core-site.xml``, ``hdfs-site.xml``).

Reviewer: ```suggestion
Comma-separated list of Hadoop configuration resource paths (for example, ``core-site.xml``, ``hdfs-site.xml``).
```
Avoid Latin abbreviations. See the GitLab documentation style guide recommended word list entry for e.g. for discussion and alternatives.
